Math Statictics Summary Distribution

In [135]:
import numpy as np
import pandas as pd

In [136]:
data = [19,29,40,22,9,12,18,100,0]
# # sum_value = sum(data)
# print(data)
# print(sum_value)
df = pd.DataFrame(data)
df.describe()
data.sort()
print(data)
df.describe()

[0, 9, 12, 18, 19, 22, 29, 40, 100]


,0
count,9.000000
mean,27.666667
std,29.466082
min,0.000000
25%,12.000000
50%,19.000000
75%,29.000000
max,100.000000


Mean, Standard Deviation , and Vairance

In [137]:
### Mean Values
mean_value = sum(data)/len(data)
print(mean_value)


27.666666666666668


In [138]:
### Standard Deviation and Variance
total = 0
for i in data:
  total += (i - mean_value)**2
variance = total/(len(data)-1) ### Sample Variance
standard_deviation = variance ** 0.5
print(f"The Standared Deviation of the Given Sample is  {standard_deviation}")
print(f"The Variance of the Given Sample is {variance}")

The Standared Deviation of the Given Sample is  29.46608219631514
The Variance of the Given Sample is 868.25


Calculating the First, Second and the third Standard Deviation Taking Positive and Negative

In [139]:
first_std = mean_value + standard_deviation
first_std1  = mean_value - standard_deviation
second_std = mean_value + (2*standard_deviation)
second_std1 = mean_value - (2*standard_deviation)
third_std = mean_value + (3*standard_deviation)
third_std1 = mean_value - (3*standard_deviation)
print(f"The First Standard Deviation is {first_std1} to {first_std}")
print(f"The Second Standard Deviation is {second_std1} to {second_std}")
print(f"The Third Standard Deviation is {third_std1} to {third_std}")

The First Standard Deviation is -1.7994155296484706 to 57.132748862981806
The Second Standard Deviation is -31.26549772596361 to 86.59883105929694
The Third Standard Deviation is -60.73157992227874 to 116.06491325561208


Calculating the Quartiles of the Data Sample with q1,q2, q3

In [140]:
### Position of the Q1, Q2, Q3
post_q1 = (len(data)+1)/4
post_q2 = (len(data)+1)/2
post_q3 = 3*(len(data)+1)/4
print(post_q1)
print(post_q2)
print(post_q3)

2.5
5.0
7.5


In [141]:
print(data)

[0, 9, 12, 18, 19, 22, 29, 40, 100]


In [142]:
q1 = (data[1]+data[2])/2
print(q1)
q2 = data[4]
print(q2)
q3 = (data[6]+data[7])/2
print(q3)

10.5
19
34.5


Interquartile Range

In [143]:
iqr = q3 - q1
print(f"The Inter Quartile Range is {iqr}")

The Inter Quartile Range is 15.5


Finding the Outliers range and Couting it's

In [144]:
## Upper OUtliers
out_upper = q3+1.5*iqr
out_upper
### Lower Outliers
out_lower = q1-1.5*iqr
out_lower
print(f"The Outliers is From the {out_lower} from {out_upper}")
count = 0
for i in data:
  if i > out_upper or i < out_lower:
    count += 1
print(f"The Number of Outliers is {count}")


The Outliers is From the -12.75 from 57.75
The Number of Outliers is 1


Central Moment ::
###
1st Central Moment
###
2nd Central Moment
###
3rd Cental Moment
###  4rd Central Moment


In [145]:
## 1st central Moment
first_sum = 0
for i in data:
  first_sum += (i-mean_value)**1
cental_moment = first_sum/(len(data)-1) ## Sample Variance
print(cental_moment) ### It Gives the Values approximately equal to the zero

-3.552713678800501e-15


In [146]:
### 2nd Central Moment
second_sum  = 0
for i in data:
  second_sum += (i-mean_value)**2
cental_moment2 = second_sum/(len(data)-1) ## Sample Variance
print(cental_moment2)

868.25


In [147]:
### 3rd Cental Moment
third_sum  = 0
for i in data:
  third_sum += (i-mean_value)**3
cental_moment3 = third_sum/(len(data)-1) ## Sample Variance
print(cental_moment3)


43383.91666666666


In [149]:
### 4th Cental Moment
fourth_sum = 0
for i in data:
  fourth_sum += (i-mean_value)**4
cental_moment4 = fourth_sum/(len(data)-1)
print(cental_moment4)

3522636.749999999


Conclusion, From the Given Data Set , It is clear that
###
The 2nd central moment provides the Sample Variance (868.25), measuring the wide dispersion of your data around the mean.

The 3rd central moment (43,383.92) indicates a strong positive skew, showing the data tail is stretched to the right by the outlier (100).

The 4th central moment (3,522,636.75) relates to Kurtosis, confirming a heavy-tailed distribution with a high probability of extreme values.